### PR TITLE
feat: add a lot of sync method, add package directory information to the project

### PR DESCRIPTION
**Pull request by amphro**

### CODE REVIEW

File: src/sfdxProject.ts [hunk 1]
```diff
@@ -42,6 +43,17 @@ export type PackageDir = {
   versionNumber?: string;
 };
 
+export type NamedPackageDir = PackageDir & {
```
Comment (amphro) [inline src/sfdxProject.ts:46]: The bulk of this file will need your review @mdonnalley

Comment (amphro) [reply src/sfdxProject.ts:46]: I'll post the PR to toolbelt when it is ready so you can see how this is used.

File: src/util/fs.ts [hunk 2]
```diff
@@ -15,8 +15,9 @@ import { promisify } from 'util';
 import { SfdxError } from '../sfdxError';
 
 type PerformFunction = (filePath: string, file?: string, dir?: string) => Promise<void>;
+type PerformFunctionSync = (filePath: string, file?: string, dir?: string) => void;
 
-export const fs = {
+export const fs = Object.assign({}, fsLib, {
```
Comment (amphro) [inline src/util/fs.ts:20]: This is what I was talking about @WillieRuemmele @nramyasri-sf - now all fs-graceful methods will be on the fs object.

File: src/keyChainImpl.ts [hunk 3]
```diff
@@ -186,7 +186,10 @@ export class KeychainAccess implements PasswordStore {
    * @param opts Options for the credential lookup.
    * @param fn Callback function (err, password).
```
Comment (shetzel) [inline src/keyChainImpl.ts:187]: callback function doc needs updating for `ConfigContents`

File: src/logger.ts [hunk 4]
```diff
@@ -274,6 +266,16 @@ export class Logger {
...
 
+  /**
+   * Create a child of the root logger, inheriting this instance's configuration such as `level`, `streams`, etc.
+   *
+   * @param name The name of the child logger.
+   * @param fields Additional fields included in all log lines.
+   */
+  public static childFromRoot(name: string, fields?: Fields): Logger {
+    return Logger.getRoot().child(name, fields);
+  }
```
Comment (shetzel) [inline src/logger.ts:277]: Woot!  🎉

File: src/logger.ts [hunk 5]
```diff
@@ -684,6 +729,44 @@ export class Logger {
     return this;
   }
 
+  public enableDEBUG() {
```
Comment (shetzel) [inline src/logger.ts:732]: Public API, so doc?

**Comment by amphro** [comment]
I pushed review comment updates and also a new feature. I probably should have done it in another PR but I was already deep in the code. See the commit diff if you need to see those changes https://github.com/forcedotcom/sfdx-core/pull/284/commits/95c415c56ba7e8a868cafb322b5481e6833da33f @shetzel

**Pull request merged**